scripts/utilities/archive_dataset_v1.py

```
@@ -0,0 +1,180 @@
+        # Check if there are uncommitted changes
+        result = subprocess.run(
+            ["git", "status", "--porcelain"], capture_output=True, text=True, cwd="."
+        )
+        has_changes = len(result.stdout.strip()) > 0 if result.returncode == 0 else True
+
+        return {
+            "commit_hash": commit_hash,
+            "branch": branch,
+            "has_uncommitted_changes": has_changes,
+        }
+    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)